src/cascadia/LocalTests_TerminalApp/TabTests.cpp

```diff
@@ -817,6 +819,159 @@ namespace TerminalAppLocalTests
         VERIFY_SUCCEEDED(result);
     }
 
+    void TabTests::MovePanes()
+    {
+        auto page = _commonSetup();
+
+        Log::Comment(L"Setup 4 panes.");
+        // Create the following layout, with ids as follows
+        // -------------------
+        // |   1    |   2    |
+        // |        |        |
+        // -------------------
+        // |   3    |   4    |
+        // |        |        |
+        // -------------------
+        auto result = RunOnUIThread([&page]() {
+            // We start with 1 tab, split vertically to get
+            // -------------------
+            // |   1    |   2    |
+            // |        |        |
+            // -------------------
+            page->_SplitPane(SplitState::Vertical, SplitType::Duplicate, 0.5f, nullptr);
+            // After this the `2` pane is focused, go back to `1` being focused
+            page->_MoveFocus(FocusDirection::Left);
+            // Split again to make the 3rd tab
+            // -------------------
+            // |   1    |        |
+            // |        |        |
+            // ---------|   2    |
+            // |   3    |        |
+            // |        |        |
+            // -------------------
+            page->_SplitPane(SplitState::Horizontal, SplitType::Duplicate, 0.5f, nullptr);
+            // After this the `3` pane is focused, go back to `2` being focused
+            page->_MoveFocus(FocusDirection::Right);
+            // Split to create the final pane
+            // -------------------
+            // |   1    |   2    |
+            // |        |        |
+            // -------------------
+            // |   3    |   4    |
+            // |        |        |
+            // -------------------
+            page->_SplitPane(SplitState::Horizontal, SplitType::Duplicate, 0.5f, nullptr);
+
+            VERIFY_ARE_EQUAL(1u, page->_tabs.Size());
+            auto tab = page->_GetTerminalTabImpl(page->_tabs.GetAt(0));
+            VERIFY_ARE_EQUAL(4, tab->GetLeafPaneCount());
+            // Our currently focused pane should be `4`
+            VERIFY_ARE_EQUAL(4u, tab->_activePane->Id().value());
+        });
+        VERIFY_SUCCEEDED(result);
+
+        // Now try to move the pane through the tree
+        Log::Comment(L"Move pane to the left. This should swap panes 3 and 4");
+        // -------------------
+        // |   1    |   2    |
+        // |        |        |
+        // -------------------
+        // |   4    |   3    |
+        // |        |        |
+        // -------------------
+        result = RunOnUIThread([&page]() {
+            // Set up action
+            MovePaneArgs args{ FocusDirection::Left };
+            ActionEventArgs eventArgs{ args };
+
+            page->_HandleMovePane(nullptr, eventArgs);
+            auto tab = page->_GetTerminalTabImpl(page->_tabs.GetAt(0));

```

Comment:
```suggestion
            page->_HandleMovePane(nullptr, eventArgs);
        }
        VERIFY_SUCCEEDED(result);
                
        result = RunOnUIThread([&page]() {    
            auto tab = page->_GetTerminalTabImpl(page->_tabs.GetAt(0));
```

Reply:
Splitting up UI calls like this gives the UI thread a chance to settle itself after modifying the UI layout.